core/dronecore.h

```diff
@@ -59,60 +56,35 @@ class DroneCore
     /**
      * @brief Translates the DroneCore::ConnectionResult enum to a human-readable English string.
      */
-    static const char *connection_result_str(ConnectionResult);
-
-    /**
-     * @brief Adds a UDP connection to the default port.
-     *
-     * This will listen on UDP port 14540.
-     *
-     * @return The result of adding the connection.
-     */
-    ConnectionResult add_udp_connection();
+    static const char *connection_result_str(const ConnectionResult &result);
 
     /**
      * @brief Adds a UDP connection to the specified port number.
      *
-     * @param local_port_number The local UDP port to listen to.
-     * @return The result of adding the connection.
-     */
-    ConnectionResult add_udp_connection(int local_port_number);
-
-    /**
-     * @brief Adds a TCP connection to the default IP address/port.
-     *
-     * This will connect to local TCP port 5760.
-     *
+     * @param local_port_number The local UDP port to listen to (defaults to 14540, the same as mavros).
      * @return The result of adding the connection.
      */
-    ConnectionResult add_tcp_connection();
+    ConnectionResult add_udp_connection(int local_port_number = 14540);
 
     /**
      * @brief Adds a TCP connection with a specific IP address and port number.
      *
-     * @param remote_ip Remote IP address to connect to.
-     * @param remote_port The TCP port to connect to.
-     * @return The result of adding the connection.
-     */
-    ConnectionResult add_tcp_connection(std::string remote_ip, int remote_port);
-
-    /**
-     * @brief Adds a serial connection with the default arguments.
-     *
-     * This will connect to serial port ttyS1 (COM0).
-     *
+     * @param remote_ip Remote IP address to connect to (defaults to 127.0.0.1).
+     * @param remote_port The TCP port to connect to (defaults to 5760).
      * @return The result of adding the connection.
      */
-    ConnectionResult add_serial_connection();
+    ConnectionResult add_tcp_connection(const std::string &remote_ip = "127.0.0.1",
+                                        int remote_port = 5760);
 
     /**
      * @brief Adds a serial connection with a specific port (COM or UART dev node) and baudrate as specified.
      *
-     * @param dev_path COM or UART dev node name/path.
-     * @param baudrate Baudrate of the serial port.
+     * @param dev_path COM or UART dev node name/path (defaults to "/dev/ttyS0").
+     * @param baudrate Baudrate of the serial port (defaults to 9600).
      * @return The result of adding the connection.
      */
-    ConnectionResult add_serial_connection(std::string dev_path, int baudrate);
+    ConnectionResult add_serial_connection(const std::string &dev_path = "/dev/ttyS0",
+                                           int baudrate = 9600);
```

Comment:
The default is super low. I'd use 57600 or 115200, I think.

Reply:
Ok. I have no clue, I just copied it from `SerialConnection` :D.